packages/web/src/components/collections-table/CollectionsTableOverflowMenuButton.tsx

```diff
@@ -0,0 +1,64 @@
+import { useGetCurrentUserId } from '@audius/common/api'
+import { Collection } from '@audius/common/models'
+import { collectionPageSelectors, CommonState } from '@audius/common/store'
+import { Flex, IconKebabHorizontal } from '@audius/harmony'
+import { useSelector } from 'react-redux'
+
+import { CollectionMenuProps } from 'components/menu/CollectionMenu'
+import Menu from 'components/menu/Menu'
+
+const { getCollection } = collectionPageSelectors
+
+type OverflowMenuButtonProps = {
+  collectionId: number
+}
+
+export const CollectionsTableOverflowMenuButton = (
+  props: OverflowMenuButtonProps
+) => {
+  const { collectionId } = props
+  const { data: currentUserId } = useGetCurrentUserId({})
+  const {
+    is_album: isAlbum,
+    playlist_owner_id: playlistOwnerId,
+    is_private: isPrivate,
+    permalink
+  } = (useSelector((state: CommonState) =>
+    getCollection(state, { id: collectionId })
+  ) as Collection) ?? {}
+
+  const overflowMenu = {
+    type: isAlbum ? ('album' as const) : ('playlist' as const),
+    playlistId: collectionId,
+    includeEmbed: true,
+    includeVisitArtistPage: false,
+    includeShare: true,
+    includeEdit: true,
+    includeSave: false,
+    isPublic: !isPrivate,
+    isOwner: currentUserId === playlistOwnerId,
+    permalink
+  } as unknown as CollectionMenuProps
```

Comment:
why the cast?

Reply:
tbh copied from [here](https://github.com/AudiusProject/audius-protocol/blob/8feb22bf3517625f0c8e83bd4786829e119a313f/packages/web/src/components/collection/desktop/OverflowMenuButton.tsx), i think we're missing some required fields from `CollectionMenuProps`.